janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -1321,24 +1320,13 @@ else if (query.getResultType() == ElementCategory.VERTEX) {
         public Iterator<JanusGraphElement> execute(final GraphCentricQuery query, final JointIndexQuery indexQuery, final Object exeInfo, final QueryProfiler profiler) {
             Iterator<JanusGraphElement> iterator;
             if (!indexQuery.isEmpty()) {
-                final List<QueryUtil.IndexCall<Object>> retrievals = new ArrayList<>();
-                // Leave first index for streaming, and prepare the rest for intersecting and lookup
-                for (int i = 1; i < indexQuery.size(); i++) {
-                    final JointIndexQuery.Subquery subquery = indexQuery.getQuery(i);
-                    retrievals.add(limit -> {
-                        final JointIndexQuery.Subquery adjustedQuery = subquery.updateLimit(limit);
-                        try {
-                            return indexCache.get(adjustedQuery,
-                                () -> QueryProfiler.profile(subquery.getProfiler(), adjustedQuery, q -> indexSerializer.query(q, txHandle).collect(Collectors.toList())));
-                        } catch (Exception e) {
-                            throw new JanusGraphException("Could not call index", e.getCause());
-                        }
-                    });
-                }
-                // Constructs an iterator which lazily streams results from 1st index, and filters by looking up in the intersection of results from all other indices (if any)
-                // NOTE NO_LIMIT is passed to processIntersectingRetrievals to prevent incomplete intersections, which could lead to missed results
-                iterator = new SubqueryIterator(indexQuery.getQuery(0), indexSerializer, txHandle, indexCache, indexQuery.getLimit(), getConversionFunction(query.getResultType()),
-                        retrievals.isEmpty() ? null: QueryUtil.processIntersectingRetrievals(retrievals, Query.NO_LIMIT));
+                // Constructs an iterator which lazily streams results if only one query is given, otherwise filters results
+                // by doing an intersection from all subQueries. When there is more than one query and adjustQuery is
+                // enabled, we must use NO_LIMIT to ensure result orders remain unchanged among executions.
```

Comment:
It seems that when backendQueryHolder.isSorted is true, we also MUST ensure the order of overall results conforms to the order the first subquery results.  Appreciate it if anyone can confirm.

Reply:
https://github.com/JanusGraph/janusgraph/blob/b2a6e8125b83038873db19ed62b43a0f6190d812/janusgraph-core/src/main/java/org/janusgraph/graphdb/query/BackendQueryHolder.java#L27-L28

This looks to me like `isSorted` causes the responsible logic to skip sorting the results as it was originally requested by the user, because we somehow know that the backend will produce the correct order on it's own. This means we are not allowed to alter the ordering of elements at any later point in time.

Please notice that although I strongly support your consideration, I can not definitively confirm it.